sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java

```diff
@@ -210,6 +214,32 @@ Mono<Response<Void>> recognizeWithResponseInternal(CallMediaRecognizeOptions rec
                     .setRecognizeOptions(recognizeOptionsInternal)
                     .setOperationContext(recognizeOptions.getOperationContext());
 
+                return contentsInternal.recognizeWithResponseAsync(callConnectionId, recognizeRequest, context);
+            } else if (recognizeOptions instanceof CallMediaRecognizeSpeechOptions) {
+                CallMediaRecognizeSpeechOptions speechRecognizeOptions = (CallMediaRecognizeSpeechOptions) recognizeOptions;
+
+                SpeechOptionsInternal speechOptionsInternal = new SpeechOptionsInternal().setEndSilenceTimeoutInMs(speechRecognizeOptions.getEndSilenceTimeoutInMs());
+
+                RecognizeOptionsInternal recognizeOptionsInternal = new RecognizeOptionsInternal()
+                    .setSpeechOptions(speechOptionsInternal)
+                    .setInterruptPrompt(speechRecognizeOptions.isInterruptPrompt())
+                    .setTargetParticipant(CommunicationIdentifierConverter.convert(speechRecognizeOptions.getTargetParticipant()));
+
+                recognizeOptionsInternal.setInitialSilenceTimeoutInSeconds((int) speechRecognizeOptions.getInitialSilenceTimeout().getSeconds());
+
+                PlaySourceInternal playSourceInternal = null;
+                if (recognizeOptions.getPlayPrompt() != null) {
+                    PlaySource playSource = recognizeOptions.getPlayPrompt();
+                    playSourceInternal = translatePlaySourceToPlaySourceInternal(playSource);
```

Comment:
I think that `convert` is a better word than `translate`. Even `getPlaySourceInternal` may work better.

Reply:
changed to convert